pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -90,13 +90,17 @@ func (p Provisioner) Provision(options *apibkt.BucketOptions) (*bktv1alpha1.Obje
 	if err != nil {
 		err = errors.Wrapf(err, "error creating bucket %q", p.bucketName)
 		logger.Errorf(err.Error())
-		p.deleteOBCResource("")
+		if err := p.deleteOBCResource(""); err != nil {
+			return nil, errors.Wrap(err, "error deleting OBCResource bucket")
```

Comment:
In the bucket provisioner, with all the calls to `deleteOBCResource()`, let's just log the error. The next line will anyway return the original error. This same call is in so many places how about a wrapper function such as:
```
func (Provisioner *p) deleteOBCResourceLogError(name string) {
  if err := p.deleteOBCResource(""); err != nil {
    logger.Warningf("error deleting OBC resource. %v", err)
  }
}
```